Develop

snakeage opened the pull request:
Описание изменений:
В этой ветке реализованы два теста:

OrderSqooterTest — проверяет успешное оформление заказа самоката до появления модального окна с подтверждением.

FaqTest — проверяет, что в блоке "Вопросы о важном" при клике на вопрос открывается соответствующий текст ответа.

Особенности:
Тесты запускаются в двух браузерах: Chrome и Mozilla Firefox.
Для переключения между браузерами и настройки окружения см. раздел README.md.

Commit:
Добавил EnvConfig с url и значениями тайм аута. В MainPage реализовал методы clickOnFaqQuestionButton д
ля поиска и скрола до вопроса в FAQ и getAnswerText возвращающего текст ответа. Реализовал класс FaqTest который проверяет что при нажатии на нужный вопрос из FAQ открывается соответсвующий ответ

Commit:
Дополнил REARME.md. В MainPage разделил кнопку заказа на две(та которая находится в хедере и та которач в середине страницы) чтобы разделить тест на два  флоу. Добавил страницу OrderPage и тесты в OrderSqooterTes.

okryukova commented on Sprint_4/README.md:
⛔️Нужно исправить. Добавь, пожалуйста, пом файл в пулл реквест

snakeage replied:
pom.xml уже добавлен в проект ранее, изменений в нем не было, поэтому он не попал в текущий PR. Проверь, пожалуйста, файл в main-ветке.

okryukova commented on Sprint_4/src/main/java/ru/practicum/pages/MainPage.java:
```diff
@@ -0,0 +1,70 @@
+    }
+
+    public void clickOnFaqQuestionButton(int index) {
+        By questionButton = By.cssSelector("#accordion__heading-" + index);
```
⛔️Нужно исправить. Локаторы нужно хранить в полях класса. Поправь, пожалуйста, во всем коде
⚠️Можно улучшить.  Для формирования локатора нужно использовать  string.format

okryukova commented on Sprint_4/src/test/java/ru/practicum/DriverFactory.java:
```diff
@@ -0,0 +1,51 @@
+    private void startChrome() {
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(IMPLICITY_TIMEOUT));
+        driver.manage().window().fullscreen();
```
⚠️Можно улучшить. Браузер, как правило, запускают в полноэкранном режиме (maximized) или определенном размере окна. Сейчас у тебя открывается в полном экране и сворачивается перед тестом

okryukova commented on Sprint_4/src/test/java/ru/practicum/FaqTest.java:
```diff
@@ -0,0 +1,57 @@
+    @Parameterized.Parameters(name = "Вопрос #{0} должен показать ответ: {1}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {0, "Сутки — 400 рублей. Оплата курьеру — наличными или картой."},
```
⚠️Можно улучшить. Тестовые данные  должны содержаться либо в константах тестового класса, либо в отдельном файле

okryukova commented on Sprint_4/src/main/java/ru/practicum/pages/MainPage.java:
```diff
@@ -0,0 +1,70 @@
+
+public class MainPage {
+    private final WebDriver driver;
+    public static final By orderButtonInHeader = By.xpath("//div[contains(@class, 'Header_Nav__AGCXC')]//button[text()='Заказать']");
```
⚠️Можно улучшить. В классическом pojo поля класса приватные.

okryukova commented on Sprint_4/src/test/java/ru/practicum/OrderSqooterTest.java:
```diff
@@ -0,0 +1,63 @@
+    @Parameterized.Parameters(name = "Тестовые данные: имя - {0}, фамилия - {1}, адрес - {2}, станция метро - {3}, номер телефона - {4}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {"Марк", "Тютчев", "кутузовсквя, 4", "чистые пруды", "89655434576", MainPage.orderButtonInHeader},
```
⛔️Нужно исправить. Не следует описывать локаторы внутри тестовых методов. Используй инкапсуляцию. В классах page objects опиши все веб-элементы, которые используются в кейсах, с помощью локаторов и укажи модификатор доступа private. Реализуй шаги для действий над веб-элементами. Тесты должны состоять из шагов (для получения актуального результата) и проверки (сравнение ожидаемого и актуального результата). Это позволит переиспользовать веб-элементы в разных шагах(методах), а шаги в разных тестах.

okryukova commented on Sprint_4/src/test/java/ru/practicum/OrderSqooterTest.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    @Test
+    public void verifyOrderSuccessMessage() {
```
⚠️Можно улучшить. В названии метода необходимо использовать слово Test. Проверь, соблюдается ли данное правило для всех созданных тестов.

Commit:
Рефакторинг:
- вынес тестовые данные для FAQ и заказов в отдельный класс TestData
- сделал все локаторы (By) и константы private final в page object
- вынес xpath-шаблоны и комментарии в статические константы
- обновил параметризованные тесты для использования централизованных данных
